python/flight-crew-scheduling/src/flight_crew_scheduling/demo_data.py

```diff
@@ -0,0 +1,255 @@
+                               times, distances)
+    # Flight assignments
+    flight_assignments = generate_flight_assignments(flights)
+    # Employees
+    employees = generate_employees(flights, dates)
+
+    # Flight Crew Schedule
+    schedule = FlightCrewSchedule(airports=airports, employees=employees, flights=flights,
+                                  flight_assignments=flight_assignments, score=None, solver_status=SolverStatus.NOT_SOLVING)
+
+    return schedule
+
+
+def generate_flights(size: int, start_datetime: datetime, airports: List[Airport],
+                     home_airports: List[Airport], dates: List[datetime.date],
+                     time_groups: List[datetime.time], distances: Dict[str, int]) -> List[Flight]:
+    if size % 2 != 0:
+        raise ValueError("The size of flights must be even")
+
+    # Departure and arrival airports
+    flights = []
+    remaining_airports = [airport for airport in airports if airport not in home_airports]
+    count_flights = 0
+
+    while count_flights < size:
+        route_size = pick_random_route_size(count_flights, size)
+        home_airport = random.choice(home_airports)
+        home_flight = Flight(flight_number=str(count_flights), departure_airport=home_airport, arrival_airport=random.choice(remaining_airports))
+        flights.append(home_flight)
+        count_flights += 1
+
+        next_flight = home_flight
+        for _ in range(route_size - 2):
+            next_flight = Flight(
+                flight_number=str(count_flights),
+                departure_airport=next_flight.arrival_airport,
+                arrival_airport=pick_random_airport(remaining_airports, next_flight.arrival_airport.code)
+            )
+            flights.append(next_flight)
+            count_flights += 1
+
+        flights.append(Flight(flight_number=str(count_flights), departure_airport=next_flight.arrival_airport, arrival_airport=home_flight.departure_airport))
+        count_flights += 1
+
+    # Assign flight numbers
+    for i, flight in enumerate(flights):
+        flight.flight_number = f"Flight {i + 1}"
+
+    # Assign flight durations
+    count_dates = size // len(dates)
+
+    def flight_consumer(f: Flight, d: date):
+        key = f"{f.departure_airport.code}-{f.arrival_airport.code}"
+        count_hours = distances[key]
+        start_time = random.choice(time_groups)
+        departure_datetime = datetime.combine(d, start_time)
+
+        if departure_datetime < start_datetime:
+            departure_datetime = start_datetime + timedelta(hours=random.randint(0, 4))
+
+        arrival_datetime = departure_datetime + timedelta(hours=count_hours)
+        f.departure_utc_date_time = departure_datetime
+        f.arrival_utc_date_time = arrival_datetime
+
+    apply_random_value_with_param(count_dates, flights, random.choice(dates),
+                       lambda f: f.departure_utc_date_time is None, flight_consumer)
```

Comment:
I personally find this hard to read/reason about. I recommend using a for loop instead:

```suggestion
    for start_date in dates:
        apply_random_value(count_dates, flights, lambda f: f.departure_utc_date_time is None, lambda f: flight_consumer(f, start_date))
```